### PR TITLE
nixos/xfconf: init

### PR DESCRIPTION
**Pull request by romildo**
###### Description of changes

- Add a module for `xfconf`, the Xfce configuration storage system.

- It was suggested in https://github.com/NixOS/nixpkgs/pull/179108#discussion_r906831062

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/programs/xfconf.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+let cfg = config.programs.xfconf;
+
+in {
+  meta = {
+    maintainers = teams.xfce.members;
+  };
+
+  options = {
+    programs.xfconf = {
```
Comment (lucasew) [inline nixos/modules/programs/xfconf.nix:13]: I think it's better if you add a `package` option that defaults to `pkgs.xfce.xfconf`, then pass that result to systemPackages and dbus packages.

**Review by lucasew — CHANGES_REQUESTED** [review]
Just one concern, the rest LGTM

Didn't test locally though

File: nixos/modules/programs/xfconf.nix (hunk 1 above)
Comment (jtojnar) [reply nixos/modules/programs/xfconf.nix:13]: Do you envision a situation where one would want to override `xfconf` for the module but not for the rest of the system the way overlays do?

Comment (romildo) [reply nixos/modules/programs/xfconf.nix:13]: Being used by many components of Xfce, it is sensible that an override of `xfconf`, if needed, would be better done system wide with overlays, for instance. So I would keep it simple and not add a `package` option.

**Pull request merged**